### PR TITLE
Fix restoring of CPENABLE

### PR DESCRIPTION
**Pull request by bugadani**
esp-wifi did not save the CPENABLE state, which sometimes resulted in the FPU being turned off. This can happen if esp-wifi decides to switch context in a lower-priority ISR, without `float-save-restore` being enabled.

The original issue is tricky to reproduce because previously we didn't change the word in the stack frame at all, so esp-wifi "restored" what was there before.

With `float-save-restore` this issue shouldn't happen since we don't disable the FPU.

### CODE REVIEW

File: hil-test/Cargo.toml [hunk 1]
```diff
@@ -254,7 +284,7 @@ generic-queue = [
 integrated-timers = [
     "esp-hal-embassy/integrated-timers",
 ]
-octal-psram = ["esp-hal/octal-psram", "dep:esp-alloc"]
+octal-psram = ["esp-hal/octal-psram", "esp-alloc"]
```
Comment (bugadani) [inline hil-test/Cargo.toml:287]: `dep:` meant there was no implicit `esp-alloc` feature, which I need in the new test.

File: hil-test/tests/esp_wifi_floats.rs [hunk 2]
```diff
@@ -0,0 +1,193 @@
+//! Cp0Disable exception regression test
+
+//% CHIPS: esp32 esp32s2 esp32s3
+//% FEATURES: esp-wifi esp-alloc
+//% FEATURES: esp-wifi esp-alloc xtensa-lx-rt/float-save-restore
```
Comment (bugadani) [inline hil-test/tests/esp_wifi_floats.rs:5]: Let's hope CI picks up the first feature set, otherwise this needs to be removed.

**Review by bjoernQ — APPROVED** [review]
Thanks! Nice you were able to create a test for this!

File: hil-test/build.rs [hunk 3]
```diff
@@ -37,5 +37,9 @@ fn main() -> Result<(), Box<dyn Error>> {
     // Define all necessary configuration symbols for the configured device:
     config.define_symbols();
 
+    if cfg!(feature = "esp-wifi") {
```
Comment (MabezDev) [inline hil-test/build.rs:40]: We should really emit this as part of esp-wifi or esp-hal's build script, I'll create an issue for that

Comment (bjoernQ) [reply hil-test/build.rs:40]: I think I tried that but it was ignored for library crates - maybe something changed since then 🤔

**Review by MabezDev — APPROVED** [review]
Thanks! I tried using this in my project where I was running into this and it now works correctly without the `float-save-restore` feature enabled :tada:

File: hil-test/build.rs (hunk 3 above)
Comment (MabezDev) [reply hil-test/build.rs:40]: Hmm yeah you're right, I forgot about that - only works in the final crate build.rs, not libs. Sorry for the noise!

**Pull request merged**